src/txn/txn_rollback_to_stable.c

```diff
@@ -1275,10 +1351,22 @@ __rollback_to_stable(WT_SESSION_IMPL *session)
 {
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
+    bool exclusive;
 
     conn = S2C(session);
+    exclusive = false;
 
-    WT_RET(__rollback_to_stable_check(session));
+    /*
+     * History store operations create transactions so we need to block eviction everywhere before
+     * doing rollback to stable's preliminary check since it checks for active transactions.
+     */
+    WT_ERR(__rollback_evict_exclusive_on(session));
```

Comment:
@kommiharibabu 
I realised that iterating over the transactions isn't thread safe. We do it to get diagnostic information because it's doesn't have to strictly be correct but we can't do that here.

At the moment, I'm just switching eviction off for the duration of the check.

Reply:
I don't think that's OK. We need to be able to evict content while rollback to stable is running, otherwise we will run out of space in the cache (or memory in the system).

Reply:
We're just turning off eviction for the check to ensure that there are no active transactions running. After this check is done, eviction is back on and rollback to stable will run.

Reply:
I see, sorry.